.github/workflows/test-integration-vizro-ai.yml

```diff
@@ -47,55 +70,76 @@ jobs:
 
   test-integration-vizro-ai:
     if: ${{ ! github.event.pull_request.head.repo.fork }}
-    name: test-integration-vizro-ai on Py${{ matrix.python-version }} ${{ matrix.label }}
-    runs-on: ubuntu-latest
+    name: test-integration-vizro-ai on Py${{ matrix.config.python-version }} ${{ matrix.config.label }}
+    runs-on: ${{ matrix.config.os }}
     strategy:
       fail-fast: false
       matrix:
-        include:
+        # This is not a nice way to define the matrix, but it's the only way that's possible so that we only run
+        # the job on Windows on merge to main rather than in every PR. See:
+        # https://stackoverflow.com/questions/66025220/paired-values-in-github-actions-matrix
+        # https://github.com/orgs/community/discussions/26253
+        # https://github.com/actions/runner/issues/1985
+        config:
           - python-version: "3.9"
             hatch-env: all.py3.9
+            os: ubuntu-latest
           - python-version: "3.10"
             hatch-env: all.py3.10
+            os: ubuntu-latest
           - python-version: "3.11"
             hatch-env: all.py3.11
+            os: ubuntu-latest
           - python-version: "3.12"
             hatch-env: all.py3.12
+            os: ubuntu-latest
           - python-version: "3.9"
             hatch-env: lower-bounds
+            os: ubuntu-latest
             label: lower bounds
+          - python-version: "3.12"
+            hatch-env: all.py3.12
+            os: windows-latest
+            label: Windows
+        is_pr:
+          - ${{ github.event_name != 'pull_request' }}
+        exclude:
+          - is_pr: true
+            config:
+              python-version: "3.12"
+              hatch-env: all.py3.12
+              os: windows-latest
+              label: Windows
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.config.python-version }}
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: ${{ matrix.config.python-version }}
 
       - name: Install Hatch
         run: pip install hatch
 
       - name: Show dependency tree
-        run: hatch run ${{ matrix.hatch-env }}:pip tree
+        run: hatch run ${{ matrix.config.hatch-env }}:pip tree
 
-      - name: Run vizro-ai integration tests with pypi vizro
-        run: |
-          export OPENAI_API_KEY=${{ secrets.OPENAI_API_KEY }}
-          export OPENAI_API_BASE=${{ secrets.OPENAI_API_BASE }}
-          export VIZRO_TYPE=pypi
-          hatch run ${{ matrix.hatch-env }}:test-integration
+      - name: Run vizro-ai integration tests with PyPI vizro
+        run: hatch run ${{ matrix.config.hatch-env }}:test-integration
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          OPENAI_API_BASE: ${{ secrets.OPENAI_API_BASE }}
+          VIZRO_TYPE: pypi
```

Comment:
FYI @l0uden this is a better way to define environment variables in GHA because it works on all OS.